pkg/framework/plugins/nodeutilization/usageclients.go

```diff
@@ -312,3 +312,23 @@ func (client *prometheusUsageClient) sync(nodes []*v1.Node) error {
 
 	return nil
 }
+
+// CollectPrometheusUsage collects the usage of all provided nodes using the
+// provided prometheus client and query. Returns a map of node names to a map
+// of resource names to their respective quantities.
+func CollectNodeUsageWithPrometheus(
+	nodes []*v1.Node, client promapi.Client, query string,
+) (map[string]map[v1.ResourceName]*resource.Quantity, error) {
+	// nopods is a filter function that always return an empty list of pods
+	// regardless of the node being queried. This is because we are only
+	// interested in the node metrics and not the pods assigned to the node.
+	nopods := func(string, podutil.FilterFunc) ([]*v1.Pod, error) {
+		return []*v1.Pod{}, nil
+	}
+
+	usageClient := newPrometheusUsageClient(nopods, client, query)
```

Comment:
You could separate the subset of the `sync` method: https://github.com/ingvagabund/descheduler/commit/d0c3a88da0628af60f4b2639b66edb28a729c4bb and have that exposed instead.

Reply:
Isn't that going to require us to expose `newPrometheusUsageClient()` as well ? Let me know what I am getting wrong here.

Reply:
OH, nevermind, I got it.

Reply:
Done.